packages/nimble-components/src/table-column/table-column-text.ts

```diff
@@ -0,0 +1,81 @@
+import { attr, css, html, ViewTemplate } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import type { StringField, TableCellState } from '../table/types';
+import { TableColumn } from './table-column';
+import {
+    bodyFontColor,
+    bodyFont,
+    controlLabelFontColor
+} from '../theme-provider/design-tokens';
+
+type TableColumnTextCellData = StringField<'value'>;
+interface TableColumnTextColumnConfig {
+    placeholder: string;
+}
+
+/**
+ * The table column for displaying strings.
+ */
+export class TableColumnText extends TableColumn<
+TableColumnTextCellData,
+TableColumnTextColumnConfig
+> {
+    public cellStateDataFieldNames = ['value'] as const;
+
+    @attr({ attribute: 'field-name' })
+    public fieldName!: string;
+
+    @attr
+    public placeholder?: string;
+
+    public override readonly cellStyles = css`
+        .placeholder {
+            font: ${bodyFont};
+            color: ${controlLabelFontColor};
+            white-space: nowrap;
+        }
+
+        .text-value {
+            font: ${bodyFont};
+            color: ${bodyFontColor};
+            white-space: nowrap;
+        }
+    `;
+
+    public readonly cellTemplate: ViewTemplate<
+    TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>
+    > = html<
+        TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>
+        >`
+        <span
+            class="${x => (x.data.value !== null && x.data.value !== undefined
+        ? 'text-value'
+        : 'placeholder')}"
+        >
+            ${x => (x.data.value !== null && x.data.value !== undefined
+        ? x.data.value
+        : x.columnConfig.placeholder)}
+        </span>
+    `;
+
+    public getColumnConfig(): TableColumnTextColumnConfig {
+        return { placeholder: this.placeholder ?? '' };
+    }
+
+    public getRecordFieldNames(): string[] {
+        if (!this.fieldName) {
+            throw new Error('valueKey must be provided.');
+        }
+        return [this.fieldName];
+    }
+
+    public validateCellData(_: TableColumnTextCellData): void {}
```

Comment:
I want to point out that despite what was shown in the spec for this column, I believe the right thing to do is allow all value types to show as string. At the very least, this column should be more lenient in its value types beyond just `string`. Things like `number` and `Date` display appropriately, and we should allow this column to be used for such values.